File: frontend/src/features/devices/schemas.ts
import { z } from "zod";

export const DeviceEntrySchema = z.object({
  id: z.string().min(1),
  name: z.string().min(1),
  ipAddress: z.string().min(1),
  createdAt: z.string().datetime({ offset: true }),
  updatedAt: z.string().datetime({ offset: true }),
});

export const DevicesResponseSchema = z.object({
  entries: z.array(DeviceEntrySchema).default([]),
});

export const DeviceCreateRequestSchema = z.object({
  name: z.string().min(1).max(128),
  ipAddress: z.string().min(1),
});

export const DeviceDeleteResponseSchema = z.object({
  status: z.string().min(1),
});

export type DeviceEntry = z.infer<typeof DeviceEntrySchema>;
export type DevicesResponse = z.infer<typeof DevicesResponseSchema>;
export type DeviceCreateRequest = z.infer<typeof DeviceCreateRequestSchema>;
export type DeviceDeleteResponse = z.infer<typeof DeviceDeleteResponseSchema>;

File: app/modules/devices/schemas.py
from __future__ import annotations

from datetime import datetime

from app.modules.shared.schemas import DashboardModel


class DeviceEntry(DashboardModel):
    id: str
    name: str
    ip_address: str
    created_at: datetime
    updated_at: datetime


class DevicesResponse(DashboardModel):
    entries: list[DeviceEntry]


class DeviceCreateRequest(DashboardModel):
    name: str
    ip_address: str


class DeviceUpdateRequest(DashboardModel):
    name: str
    ip_address: str


class DeviceDeleteResponse(DashboardModel):
    status: str
